### DISCUSSION TITLE
Accessing files in the public directory programmatically

### DISCUSSION THREAD

**Discussion by PseudoMon** [remix-run/remix#2302] [Q&A]
Hi! I'm wondering if there's any way to access files in the `public` directory directly through server-side JavaScript? That is, using `fs` instead of, say, in the HTML through `<img src="/something.png">`. My use case is that there's a list of images I want to show, but in case an image doesn't exist in the directory (which I was trying to find out through `fs.readdir(...).includes(...)`),  I want to replace the image's URL with a fallback URL, and I'd like for all this to happen server-side instead of waiting for the image to fail to load in the browser.

Another use case is that I want to keep some data that my site uses in publicly-available JSON files. I know you can just `import` JSON file if it's located inside the `app` folder like the rest of the source code, but I'd like to keep this data separate from the site's source and also accessible to anyone by entering its URL. 

I've tried straight up accessing `fs.readdir("public/.....")`, and that works when running locally, but threw an `ENOENT: no such file or directory` error when I tried to deploy it to Netlify.

**Comment by frontalicious-martijn** [comment]
This is probably no longer relevant but have you tried `fs.readDir("./public/...")`? Since you are on the server you have to use paths are you would in a terminal.

**Reply by PseudoMon** [reply]
It actually still is since I recently started a new project with similar use case! This doesn't work, unfortunately. I've also tried  using `process.cwd()` instead of `.` but to no avail.

Netlify/Vercel/Fly.io seems to place the files arbitrarily, so there's just no way to figure out where anything is relative to where the process is run.
